core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelSet.java

```diff
@@ -32,6 +34,15 @@
  */
 @ThreadSafe
 class ChannelSet implements Iterable<DriverChannel> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ChannelSet.class);
+  /**
+   * The maximum number of iterations in the busy wait loop in {@link #next()} when there are
+   * multiple channels. This is a backstop to protect against thread starvation, in practice we've
+   * never observed more than 3 iterations in tests.
+   */
+  private static final int MAX_ITERATIONS = 50;
```

Comment:
maybe we should add a unit test for that `MAX_ITERATIONS`?

Reply:
Right, I think it can be covered by a test.